### DISCUSSION TITLE
Automated ~/Lists publishing when changed

### DISCUSSION THREAD

**Discussion by rubiojr** [neurosnap/lists.sh#24] [Show and tell]
I use a dependency free (using built-in Linux distribution tools, that is) mechanism to automatically publish `*.txt` changes when I edit them.

In a nutshell, [this script](https://github.com/rubiojr/scripts/blob/master/lists-sh-update.sh) will use a couple of systemd user units to `scp` `.txt` files when there are changes in `$HOME/Lists`:

```bash
#!/bin/bash
#
# Automatically scp files to https://lists.sh
# when $HOME/Lists files are changed.
#
# Setup a lists.sh account first (See https://lists.sh)
#
# Usage:
#
#   * Setup a lists.sh account first (See https://lists.sh)
#   * Update the LISTS_PATH variable if you don't want your lists to live in ~/Lists
#   * Run this script
#
set -e

### CONFIGURATION ###

LISTS_PATH="$HOME/Lists"  # Directory holding .txt files
NOTIFY_SEND=1 # Comment out this variable if you don't want notifications

#####################

SDPATH="$HOME/.config/systemd/user"

mkdir "$SDPATH" "$LISTS_PATH" -p

cat > "$SDPATH/lists-sh.path" <<EOF
[Unit]
Description=lists.sh directory

[Path] 
PathChanged=%h/Lists
Unit=lists-sh.service

[Install]
WantedBy=default.target
EOF

cat > "$SDPATH/lists-sh.service" <<EOF
[Unit]
Description=Update lists.sh

[Service]  
Type=simple  
ExecStart=%h/.config/systemd/user/lists-sh.script
EOF

cat > "$SDPATH/lists-sh.script" <<EOF
#!/bin/sh
scp ~/Lists/*txt lists.sh:
if [ -n "$NOTIFY_SEND" ]; then
  notify-send "lists.sh content updated!"
fi
EOF
chmod +x "$SDPATH/lists-sh.script"

systemctl --user daemon-reload
systemctl --user enable lists-sh.path

```

**Comment by neurosnap** [comment]
This is awesome!  Thanks for sharing!  I'm going to link to this discussion in the lists.sh help page.

**Comment by rubiojr** [comment]
@neurosnap I haven't found a way to remove a list/file without using the TUI. Is that something we can do? It'd be nice to improve the script so it removes remote files when files are deleted locally.

**Reply by neurosnap** [reply]
I felt that since `scp` doesn't natively support removing files on the remote system that it would be weird to permit it for lists.sh.  Also, I thought of situations where you would want to just publish a new post without sending all of your posts.

Having said that, if you send a file with 0 bytes then we will delete the item from the system.
